### PR TITLE
Fail fast when contained dies in local-run.  Resolves #213

### LINKED ISSUE #213: If container dies, fail healthcheck (local-run)
If the container is not running at all, it should stop polling for a passing healthcheck and terminate immediately.

Here's a run that could have been terminated very early on (caused by a typo):

```
$ sudo paasta local-run -p -i main -s sessions -c pnw-stagea
Please wait while the image (...) is pulled...
...

Digest: sha256:8f654c56cca4a6db68de47efc93d873e719e64fa6964c1adf5262703024e7ce5
Status: Downloaded newer image for ...
You're running a container in non-interactive mode.
This is how Mesos runs containers.
Note that some programs behave differently when running with no tty attached (as your program is about to run).

Running docker command:
docker run ...
Found our container running with CID 97eac2cd99fa7ed4416c48f4f5ef7b07e61e0eb21d8cf4c2525e49c3c04542da

Starting health check via http://stageaservices1.sldev.yelpcorp.com:55405/status (waiting 60 econds before considering failures due to grace period):
Healthcheck failed (disregarded due to grace period)
Healthcheck failed (disregarded due to grace period)
Healthcheck failed (disregarded due to grace period)
Healthcheck failed (disregarded due to grace period)
Healthcheck failed (disregarded due to grace period)
Healthcheck failed (disregarded due to grace period)
Healthcheck failed! (Attempt 1 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 2 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 3 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 4 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 5 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 6 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 7 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 8 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 9 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 10 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 11 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 12 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 13 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 14 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 15 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 16 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 17 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 18 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 19 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 20 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 21 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 22 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 23 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 24 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 25 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 26 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 27 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 28 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 29 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Healthcheck failed! (Attempt 30 of 30) (via: http://stageaservices1.sldev.yelpcorp.com:55405/status)
Your service is now running! Tailing stdout and stderr:
[: 1: bash,: unexpected operator

^CCan't run Docker container. Error: 404 Client Error: Not Found ("no such id: 97eac2cd99fa7ed4416c48f4f5ef7b07e61e0eb21d8cf4c2525e49c3c04542da")
```

### PR DESCRIPTION
**Pull request by asottile**

### CODE REVIEW

File: tests/cli/test_cmds_local_run.py [hunk 1]
```diff
@@ -943,21 +943,21 @@ def test_simulate_healthcheck_on_service_enabled_during_grace_period(
...
-    # First run_healthcheck_on_container call happens silently
-    assert mock_run_healthcheck_on_container.call_count == 2
+    assert mock_run_healthcheck_on_container.call_count == 1
 
 
 @mock.patch('time.sleep', autospec=True)
 @mock.patch('time.time', autospec=True)
 @mock.patch('paasta_tools.cli.cmds.local_run.run_healthcheck_on_container',
-            autospec=True, return_value=[False, True])
+            autospec=True, side_effect=[False, False, True])
```
Comment (asottile) [inline tests/cli/test_cmds_local_run.py]: this test was actually completely wrong! my changes ticked it so I fixed it

File: tests/cli/test_cmds_local_run.py [hunk 2]
```diff
@@ -975,8 +975,37 @@ def test_simulate_healthcheck_on_service_enabled_honors_grace_period(
...
     assert simulate_healthcheck_on_service(
         mock_service_manifest, mock_docker_client, fake_container_id, fake_mode, fake_url, True)
-    assert mock_sleep.call_count == 0
-    assert mock_run_healthcheck_on_container.call_count == 2
+    assert mock_sleep.call_count == 2
+    assert mock_run_healthcheck_on_container.call_count == 3
+    out, _ = capsys.readouterr()
+    assert out.count('Healthcheck failed! (disregarded due to grace period)') == 1
+    assert out.count('Healthcheck failed! (Attempt') == 1
+    assert out.count('Healthcheck succeeded!') == 1
```
Comment (asottile) [inline tests/cli/test_cmds_local_run.py]: now with stronger assertions

**Comment by asottile** [comment]
I also did some local testing with this branch:

## Fast exiting
```
$ cli.py local-run -s www_pages
Local Makefile with 'cook-image' target detected. Assuming --build
Guesing cluster configuration for norcal-devb. To override, use the --cluster option.
Guessing instance configuration for main. To override, use the --instance option.
docker build -t paasta-local-run-www_pages-asottile .
Sending build context to Docker daemon 37.89 kB
Sending build context to Docker daemon 
Step 0 : FROM ubuntu:latest
 ---> 8693db7e8a00
Step 1 : CMD bash -exc echo RUNNING && exit 1
 ---> Using cache
 ---> 24dea624a3c6
Successfully built 24dea624a3c6
You're running a container in non-interactive mode.
This is how Mesos runs containers.
Note that some programs behave differently when running with no
tty attached (as your program is about to run).

Running docker command:
docker run <snip>
Found our container running with CID 93ffa70565dc494b0414e339c45e0a101b6f4de184647866a94a086397cedd52

Starting health check via http://dev25-devc.dev.yelpcorp.com:45028/status (waiting 60 seconds before considering failures due to grace period):
Healthcheck failed! (disregarded due to grace period)
Container exited with code 1
Your service failed to start, here is the stdout and stderr
+ echo RUNNING
+ exit 1
RUNNING

Stopping and removing the old container 93ffa70565dc494b0414e339c45e0a101b6f4de184647866a94a086397cedd52...
(Please wait or you may leave an orphaned container.)
...done
```

## Slow exiting
```
$ cli.py local-run -s www_pages
Local Makefile with 'cook-image' target detected. Assuming --build
Guesing cluster configuration for norcal-devb. To override, use the --cluster option.
Guessing instance configuration for main. To override, use the --instance option.
docker build -t paasta-local-run-www_pages-asottile .
Sending build context to Docker daemon 37.89 kB
Sending build context to Docker daemon 
Step 0 : FROM ubuntu:latest
 ---> 8693db7e8a00
Step 1 : CMD bash -exc echo RUNNING && sleep 65 && exit 0
 ---> Using cache
 ---> 0b54b2d90e42
Successfully built 0b54b2d90e42
You're running a container in non-interactive mode.
This is how Mesos runs containers.
Note that some programs behave differently when running with no
tty attached (as your program is about to run).

Running docker command:
docker run <snip>
Found our container running with CID c7d0ced2c1d83219a7e17831b9ef22d59db67a4e503a70878f4917505988c91d

Starting health check via http://dev25-devc.dev.yelpcorp.com:53121/status (waiting 60 seconds before considering failures due to grace period):
Healthcheck failed! (disregarded due to grace period)
Healthcheck failed! (disregarded due to grace period)
Healthcheck failed! (disregarded due to grace period)
Healthcheck failed! (disregarded due to grace period)
Healthcheck failed! (disregarded due to grace period)
Healthcheck failed! (disregarded due to grace period)
Healthcheck failed! (Attempt 1 of 30) (via: http://dev25-devc.dev.yelpcorp.com:53121/status)
Container exited with code 0
Your service failed to start, here is the stdout and stderr
+ echo RUNNING
+ sleep 65
RUNNING
+ exit 0

Stopping and removing the old container c7d0ced2c1d83219a7e17831b9ef22d59db67a4e503a70878f4917505988c91d...
(Please wait or you may leave an orphaned container.)
...done
```

**Comment by solarkennedy** [comment]
This looks great! Thanks so much!

**Pull request merged**